pkgs/development/python-modules/automx2/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+, flask
+, flask-migrate
+, ldap3
+}:
+buildPythonPackage rec {
+  pname = "automx2";
+  version = "2022.1";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-My0YT5FLYjXSRKC0oUYNybewv6CIKr93tqMZ2fYY+5I=";
+  };
+
+  propagatedBuildInputs = [
+    flask
+    flask-migrate
+    ldap3
+  ];
+
```

Comment:
Could you please enable checkPhase and pythonImportsCheck?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

Reply:
I'd like to but both the approaches given there seem to error out in somewhat mysterious ways.  If I use

```nix
  nativeCheckInputs = [ pytest ];
  checkPhase = ''
    runHook preCheck
    pytest
    runHook postCheck
  '';
```

I get
```
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0
rootdir: /build/automx2-2022.1
collected 0 items                                                              

============================ no tests ran in 0.00s =============================
error: builder for '/nix/store/c9rxp7m26fx7sm1259mpsnc8iplxdwkz-python3.10-automx2-2022.1.drv' failed with exit code 5;
       last 10 log lines:
       > patching script interpreter paths in /nix/store/n7mbvy8sqlninhgkciq3kzia66pqiv9j-python3.10-automx2-2022.1-dist
       > Executing pythonRemoveTestsDir
       > Finished executing pythonRemoveTestsDir
       > running install tests
       > ============================= test session starts ==============================
       > platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0
       > rootdir: /build/automx2-2022.1
       > collected 0 items                                                              
       >
       > ============================ no tests ran in 0.00s =============================
       For full logs, run 'nix log /nix/store/c9rxp7m26fx7sm1259mpsnc8iplxdwkz-python3.10-automx2-2022.1.drv'.
```

Whereas if I try to use `pytestCheckHook` in `nativeCheckInputs`, I get:

```
       > Executing pytestCheckPhase
       > ============================= test session starts ==============================
       > platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0
       > rootdir: /build/automx2-2022.1
       > collected 0 items                                                              
       >
       > ============================ no tests ran in 0.00s =============================
       > /nix/store/vr6wwdxkmyy44sg0gwxi10b8fc5zhwz0-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/mh6wg0ls8bfwzqwm9b7h88zvr4pk8syy-python3.10-automx2-2022.1.drv'.
```

If you have any ideas as to how to fix this I'd be appreciative, otherwise I'll have to have a look at them in the morning.

`pythonImportsCheck` works fine though.

Reply:
Oh it seems to simply be that the PyPI distribution doesn't ship with any tests — 5 is the pytest code for ‘no tests found’.  I'll just enable `pythonImportsCheck`.

Reply:
Please consider using fetchFromGitHub to improve package reliability.

Reply:
Okay, this version (the latest stable) doesn't pass tests due to some setup issues, but the latest master does, so I've bumped the version to that so we can have tests.